zstacklib/zstacklib/utils/file_metadata_handler.py

```
@@ -0,0 +1,356 @@
+        with self._lock_map_lock:
+            if metadataPath not in self._lock_map:
+                self._lock_map[metadataPath] = threading.Lock()
+                if len(self._lock_map) > self._LOCK_MAP_HIGH_WATER:
+                    self._evict_unlocked_entries()
+            return self._lock_map[metadataPath]
+
+    def _evict_unlocked_entries(self):
+        """Remove entries whose locks are not currently held.
+        Called with _lock_map_lock already held."""
+        to_remove = []
+        for path, lock in self._lock_map.items():
```

Ruff (0.15.7)

[error] Import `lock` from line 7 shadowed by loop variable

(F402)